packages/plugin-vue/src/prebundle.ts

```diff
@@ -0,0 +1,153 @@
+import { readFileSync } from 'node:fs'
+import { dirname } from 'node:path'
+import type { DepOptimizationOptions, ResolvedConfig, UserConfig } from 'vite'
+import { createFilter, normalizePath } from 'vite'
+import { transformMain } from './main'
+import { EXPORT_HELPER_ID, helperCode } from './helper'
+import type { ResolvedOptions } from './index'
+
+type ESBuildPlugin = NonNullable<
+  NonNullable<DepOptimizationOptions['esbuildOptions']>['plugins']
+>[0]
+
+const FACADE_PLUGIN: ESBuildPlugin = {
+  name: 'vite-plugin-vue:facade-prebundle',
+  setup: () => {},
+}
+
+const PLUGIN_NAME = 'vite-plugin-vue:prebundle'
+
+export function createOptimizeDeps(
+  config: UserConfig,
+  options: ResolvedOptions,
+): DepOptimizationOptions | undefined {
+  if (!options.prebundleSfc) {
+    return config.optimizeDeps
+  }
+
+  const nextOptimizeDeps = (config.optimizeDeps ||= {})
+  const exts = (nextOptimizeDeps.extensions ||= [])
+  exts.push('.vue')
```

Comment:
Skip if exists

Reply:
resolve in b5cf2a90cc26918f3af12338389b73bf30fcbb2b